app/src/main/java/org/mozilla/focus/FocusApplication.kt

```diff
@@ -18,6 +18,8 @@ class FocusApplication : LocaleAwareApplication() {
     lateinit var visibilityLifeCycleCallback: VisibilityLifeCycleCallback
         private set
 
+    val components by lazy { Components() }
```

Comment:
It's `lazy` because at the time `FocusApplication` gets created it's not a valid `Context` yet that can be used as a `Context` for various things. So I delay the creation until it's actually needed.

Reply:
I think I'd prefer `lateinit` with initialization in `onCreate`: that way it's explicit after which point it's intended to be used.

But if you prefer lazy, please add a comment explaining why it's necessary (not a valid Context yet).